lib/elixir/test/elixir/kernel/raise_test.exs

```diff
@@ -361,6 +361,34 @@ defmodule Kernel.RaiseTest do
       assert result == "expected a function, got: :example"
     end
 
+    test "badfun error when the function is gone" do
+      defmodule Foo do
```

Comment:
Instead of calling it `Foo`, which has a high chance of conflicts, maybe we should call it `BadFunction`? Thoughts?